src/script/lua_api/l_object.cpp

```diff
@@ -2220,26 +2184,18 @@ int ObjectRef::l_get_day_night_ratio(lua_State *L)
 	return 1;
 }
 
-// set_minimap_modes(self, modes, modeindex)
+// set_minimap_modes(self, modes, selected_mode)
 int ObjectRef::l_set_minimap_modes(lua_State *L)
 {
 	NO_MAP_LOCK_REQUIRED;
-
-	// Arg 1 should be a player
 	ObjectRef *ref = checkobject(L, 1);
 	RemotePlayer *player = getplayer(ref);
-	if (player == NULL)
+	if (player == nullptr)
 		return 0;
 
-	// Arg 2 should be a table (of tables)
-	if (!lua_istable(L, 2)) {
-		return 0;
-	}
-
-	// Arg 3 should be a number
-	s16 wantedmode = lua_tonumber(L, 3);
-
+	luaL_checktype(L, 2, LUA_TTABLE);
 	std::vector<MinimapMode> modes;
+	s16 selected_mode = luaL_checkint(L, 3);
```

Comment:
`"2"` is a valid number but fails in `luaL_checkint`.

Reply:
(Missing type coercion)